ARAKOON-360: Allow multiple 'preferred master' nodes

NicolasT opened the pull request.

NicolasT committed:
Turn 'preferred master' into 'string list'

In order to support configuration of multiple 'preferred master' nodes,
we should be able to track multiple names instead of one.

A subsequent commit will introduce support for this in through
configuration.

See: ARAKOON-360
See: http://jira.incubaid.com/browse/ARAKOON-360

NicolasT committed:
Enable configuration of multiple preferred masters

This commit adds a new configuration item, 'preferred_masters', which
works similar to the 'master' & 'preferred_master' combinations, but
allows setting a list of preferred master nodes.

See: 0c064f3eaa7f0eaf38ae78ac3fcbc4007020226d
See: ARAKOON-360
See: http://jira.incubaid.com/browse/ARAKOON-360

NicolasT committed:
Extension: Add 'preferred_masters' configuration feature

See: 98d74321f5d7c2f54d5faaba85f379b2a572c35c
See: ARAKOON-360
See: http://jira.incubaid.com/browse/ARAKOON-360

toolslive commented:
Where is the (unit|system) test that 
- the way we configure things with the qshell extension is not broken.
- shows Arakoon behaves as it should?

NicolasT committed:
Test: Test 'preferredMasters' extension method

See: d8a6072cbaf5ef66b5229035c6a60c97b3509347
See: ARAKOON-360
See: http://jira.incubaid.com/browse/ARAKOON-360

NicolasT committed:
Tests: Add first test for 'preferred_masters'

This commit adds a first test for 'preferred_masters', by copying the
existing 'system_tests_preferred' test, and changing it slightly to use
'preferredMasters' instead of 'forceMaster' with 'preferred=True'.

See: 98d74321f5d7c2f54d5faaba85f379b2a572c35c
See: d8a6072cbaf5ef66b5229035c6a60c97b3509347
See: ARAKOON-360
See: http://jira.incubaid.com/browse/ARAKOON-360

NicolasT committed:
Tests: Add 'preferred masters' failover test

See: e90d0c585dde96a780dbd1f4109d03dba651b9c2
See: 98d74321f5d7c2f54d5faaba85f379b2a572c35c
See: d8a6072cbaf5ef66b5229035c6a60c97b3509347
See: ARAKOON-360
See: http://jira.incubaid.com/browse/ARAKOON-360

NicolasT commented:
Added tests. Also ran some more manual validation of failover behavior, seems to work as desired/expected.

Commit:
fixed typo
